src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private StorageDao storageDao;
+    private User actualuser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
+        actualuser = new User();
+        actualuser.setLogin("sidorov");
+        actualuser.setPassword("123456");
+        actualuser.setAge(35);
+    }
+
+    @Test
+    void age_18_Ok() {
+        actualuser.setAge(18);
+        User expectedUser = registrationService.register(actualuser);
+        assertEquals(expectedUser, actualuser);
+        assertEquals(storageDao.get(expectedUser.getLogin()), actualuser);
+        assertEquals(Storage.people.size(), 1);
+    }
+
+    @Test
+    void age_more18_Ok() {
+        actualuser.setAge(29);
+        User expectedUser = registrationService.register(actualuser);
+        assertEquals(expectedUser, actualuser);
+        assertEquals(storageDao.get(expectedUser.getLogin()), actualuser);
+        assertEquals(Storage.people.size(), 1);
+    }
+
+    @Test
+    void age_less18_notOk() {
+        actualuser.setAge(17);
+        assertThrows(RuntimeException.class, () -> registrationService.register(actualuser));
+    }
+
+    @Test
+    void age_null_notOk() {
+        actualuser.setAge(null);
+        assertThrows(RuntimeException.class, () -> registrationService.register(actualuser));
+    }
+
+    @Test
+    void age_0_notOk() {
+        actualuser.setAge(0);
+        assertThrows(RuntimeException.class, () -> registrationService.register(actualuser));
+    }
+
+    @Test
+    void age_less0_notOk() {
+        actualuser.setAge(-25);
+        assertThrows(RuntimeException.class, () -> registrationService.register(actualuser));
+    }
+
+    @Test
+    void age_maxint_Ok() {
+        actualuser.setAge(2147483647);
+        User expectedUser = registrationService.register(actualuser);
+        assertEquals(expectedUser, actualuser);
+        assertEquals(storageDao.get(expectedUser.getLogin()), actualuser);
+        assertEquals(Storage.people.size(), 1);
```

Comment:
The test case `age_maxint_Ok` is testing an unrealistic age value. It is not necessary to test for the maximum integer value of age. Remove this test to avoid confusion and focus on realistic scenarios.